packages/rest/src/router/routing-table.ts

```diff
@@ -148,11 +151,26 @@ export class RoutingTable {
    * @param request
    */
   find(request: Request): ResolvedRoute {
-    for (const entry of this._routes) {
-      const match = entry.match(request);
-      if (match) return match;
+    debug('Finding route %s for %s %s', request.method, request.path);
+    const method = request.method.toLowerCase();
+    const path = `/${method}${request.path}`;
+    try {
+      const found = this._router.match(path);
+      debug('Route matched: %j', found);
+      const route = this._routes.find(r => {
+        // Normalize the path
+        const p = r.path.startsWith('/') ? r.path : `/${r.path}`;
+        return method === r.verb && `/${method}${p}` === found.route;
+      });
+      if (route) {
+        debug('Route found: %s', inspect(route, {depth: 5}));
+        return createResolvedRoute(route, found.params);
+      }
+    } catch (err) {
+      // Ignore the error, will report as NotFound
```

Comment:
Please add a debug log to make troubleshooting easier when something unexpected happens inside wayfarer.